syntax = "proto3";

package libp2p.peer.pb.crypto;

option go_package = "github.com/libp2p/go-libp2p/core/crypto/pb";

enum KeyType {
    RSA = 0;
    Ed25519 = 1;
    Secp256k1 = 2;
    ECDSA = 3;
}

message PublicKey {
    KeyType Type = 1;
    bytes Data = 2;
}

message PrivateKey {
    KeyType Type = 1;
    bytes Data = 2;
}
